storybookjs/storybook
# Docs: Show source of corresponding args in source block

Closes #20745

## What I did

- Emit args with `SNIPPET_RENDERED` event
- Store code in source container keyed on args hash
- Read code based on args except if `__forceInitialArgs` is passed to `Source` (in which case read on `initialArgs`).
- Pass `__forceInitialArgs` through.

## How to test

See stories & E2E tests. Try changing args on docs page with source open and check only the primary story changes.



## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
